### PR TITLE
Don't override runtime npm packages when user provides their own

### PR DESCRIPTION
**Pull request by eweiter**
See #58 

When a user inherits from our runtimes and they run an npm install with their own package.json it will clobber our npm modules. 

The solution was to move the runtime's npm modules to the root of the Docker container so that when a user installs their own they will be at the user level of the container and if the runtime cannot find the npm module in the user level it will default up to trying to find it from the root level.

### CODE REVIEW

**Comment by eweiter** [comment]
@rabbah Since @csantanapr is on vacation I'm not sure if you want to review this PR?

File: tests/src/test/scala/runtime/actionContainers/NodeJsActionContainerTests.scala [hunk 1]
```diff
@@ -585,4 +586,40 @@ abstract class NodeJsActionContainerTests extends BasicActionRunnerTests with Ws
         e shouldBe empty
     })
   }
+
+  it should "not override runtime packages with user provided packages" in {
+    withContainer(nodejsTestDockerImageName) { c =>
+      val code = """
+                   | function main(args) {
+                   |  var ow2 = require('openwhisk');
+                   |  var ow3 = require('../node_modules/openwhisk')
```
Comment (csantanapr) [inline tests/src/test/scala/runtime/actionContainers/NodeJsActionContainerTests.scala]: This test has two concerns, I think the test should validate that openwhisk 2.x is possible.
And there are two flavors via Dockerfile, which this one covers, and a new test via zip.
And not test the existence of of v3, maybe in another test to test but not 100% sure.

**Comment by eweiter** [comment]
@csantanapr 
This should be ready, it has the 3 tests we talked about
1) (Already existing) , check that the "openwhisk" package is available
2) In a Docker file a user should be able to require an older version of the "openwhisk" package
3) In a .zip file a user should be able to require an older version of the "openwhisk" package

File: tests/src/test/scala/runtime/actionContainers/NodeJs8ActionContainerTests.scala [hunk 2]
```diff
@@ -19,12 +19,13 @@ package runtime.actionContainers
 
 import org.junit.runner.RunWith
 import org.scalatest.junit.JUnitRunner
-import spray.json.JsObject
+import spray.json.{JsObject}
```
Comment (csantanapr) [inline tests/src/test/scala/runtime/actionContainers/NodeJs8ActionContainerTests.scala]: No need for `{}` if it's only one class

File: tools/travis/build.sh [hunk 3]
```diff
@@ -38,8 +38,16 @@ TERM=dumb ./gradlew install
...
+# Build runtime and dependencies
 cd $ROOTDIR
+
+# Build runtime
 TERM=dumb ./gradlew \
 :core:nodejs6Action:distDocker \
 :core:nodejs8Action:distDocker
+
+# Build test dependency containers
+docker build --tag nodejs6docker ./tests/dat/docker/nodejs6docker/
```
Comment (csantanapr) [inline tools/travis/build.sh]: Don't build the images manually use a gradle project and the `distDocker` task

File: tests/src/test/scala/runtime/actionContainers/NodeJsActionContainerTests.scala [hunk 4]
```diff
@@ -586,4 +589,44 @@ abstract class NodeJsActionContainerTests extends BasicActionRunnerTests with Ws
...
+                   |
+                   |  return { "message": "success" };
+                   |}
+                 """.stripMargin
+
+      val (initCode, err) = c.init(initPayload(code))
+
+      initCode should be(200)
+
+      // WHEN I run an action that requires ws and socket.io.client
```
Comment (csantanapr) [inline tests/src/test/scala/runtime/actionContainers/NodeJsActionContainerTests.scala]: I don't understand this comment about `ws` and `socket.io`

File: tools/travis/build.sh (hunk 3 above)
Comment (csantanapr) [reply tools/travis/build.sh]: you can create two project and include them, then just run `./gradlew distDocker` it will build all 4 images

**Pull request merged**